bundles/org.openhab.binding.iammeter/src/main/java/org/openhab/binding/iammeter/internal/IammeterHandler.java

```diff
@@ -0,0 +1,192 @@
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.iammeter.internal;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import javax.measure.Unit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.thing.binding.builder.ThingBuilder;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.eclipse.smarthome.io.net.http.HttpUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.google.gson.JsonSyntaxException;
+
+/**
+ * The {@link IammeterHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Yang Bo - Initial contribution
+ */
+
+@NonNullByDefault
+public class IammeterHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(IammeterHandler.class);
+    private @Nullable ScheduledFuture<?> refreshJob;
+    private IammeterConfiguration config;
+    private static final int TIMEOUT_MS = 5000;
+
+    public IammeterHandler(Thing thing) {
+        super(thing);
+        config = getConfiguration();
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            refresh();
+        }
+    }
+
+    private boolean bExtraChannelRemoved = false;
+
+    @Override
+    public void initialize() {
+        ScheduledFuture<?> refreshJob = this.refreshJob;
+        config = getConfiguration();
+        if (refreshJob == null) {
+            refreshJob = scheduler.scheduleWithFixedDelay(this::refresh, 0, config.refreshInterval, TimeUnit.SECONDS);
+            updateStatus(ThingStatus.UNKNOWN);
+        }
+    }
+
+    private void refresh() {
+        IammeterConfiguration config = this.config;
+        try {
+            String httpMethod = "GET";
+            String url = "http://admin:admin@" + config.host + ":" + config.port + "/monitorjson";
```

Comment:
Is the user/password configurable? In that case it should be a thing parameter. Also it's better to pass it as a header parameter instead of putting in the url.

Reply:
it's configurable now.